include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,330 @@
+                          | ranges::view::chunk(3)
+                          | ranges::view::transform(translate);
+        }
+        else
+        {
+            return irange  | ranges::view::chunk(3)
+                           | ranges::view::take_while([] (auto chnk) { return ranges::size(chnk) == 3; })
+                           | ranges::view::transform(translate);
+        }
+    }
+
+    /*!\brief Pipe operator that enables view-typical use of pipe notation.
+    * \tparam irng_t The type of the range being processed.
+    * \param irange The range being processed as left argument of the pipe.
+    * \param translate_single_fn_v The result of the single-argument operator().
+    */
+    template <typename irng_t>
+    friend auto operator|(irng_t && irange, seqan3::detail::translate_single_fn const & translate_single_fn_v)
+    {
+        return translate_single_fn_v(std::forward<irng_t>(irange));
+    }
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\brief A view performs single frame translation of nucleotide into aminoacid alphabet
+ * \tparam irng_t The type of the range being processed.
+ * \param irange The range being processed.
+ * \returns A range containing aminoacid sequence. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to translate nucleotide sequences into single forward aminoacid sequences.
+
+ * \par View properties
+ *
+ * |                     | `irng_t` (range input type)           | `rrng_t` (range return type)                              |
+ * |---------------------|---------------------------------------|-----------------------------------------------------------|
+ * | range               | seqan3::input_range_concept           | seqan3::view_concept + all range concepts met by `irng_t` |
+ * | `range_reference_t` | seqan3::nucleotide_concept            |                                                           |
+ *
+ * * `irng_t` is the type of the range modified by this view (input).
+ * * `rrng_type` is the type of the range returned by this view.
+ * * for more details, see \ref view.
+ *
+ * \par Example
+ *
+ * Operating on a range of seqan3::dna5:
+ * ```cpp
+ * dna5_vector vec{"ACGTACGTACGTA"_dna5};
+ *
+ * // single frame translation
+ * auto v1 = vec | view::translate_single;                     // == [T,Y,V,R]
+ *
+ * // function syntax
+ * auto v2 = view::translate_single(vec);                      // == [T,Y,V,R]
+ *
+ * // combinability
+ * auto v3 = vec | view::complement | view::translate_single;  // == [C,M,H,A]
+ * ```
+ */
+seqan3::detail::translate_single_fn const translate_single;
+} // namespace seqan3::view
+
+
+namespace seqan3
+{
+/*!\brief Specialization values for translation frames.
+ * \details The numeric values of the enums correspond to the number of frames that is represented.
+ */
+enum translation_frames
```

Comment:
This should be a scoped enum, i.e ` enum class ...`. This also means you need to prefix your enum usage, but it improves readability and reduces naming conflicts.